### PR TITLE
MetadataApi from CoW SDK

### PR DESCRIPTION
**Pull request by shoom3301**
Moved the whole metadata API:
https://github.com/cowprotocol/cow-sdk/blob/c3f72fd7100dfe774c656afc006cd00db8b56df4/src/api/metadata/index.ts#L19

### CODE REVIEW

**Comment by shoom3301** [comment]
@SocketSecurity ignore protobufjs@6.11.3

File: src/utils.ts [hunk 1]
```diff
@@ -52,7 +52,7 @@ export async function getAppDataSchema(version: string): Promise<AnyAppDataDocVe
     throw new Error(`AppData version ${version} is not a valid version`)
   }
   try {
-    return await import(`../schemas/v${version}.json`)
+    return await import(`../schemas/v${version}.json`).then(m => m.default)
```
Comment (alfetopito) [inline src/utils.ts:55]: Why is this needed?

**Review by alfetopito — APPROVED** [review]
Have you tested this anywhere yet, like the Explorer?

File: package.json [hunk 2]
```diff
@@ -36,20 +36,35 @@
...
     "ts-node": "^10.8.2",
-    "typescript": "^4.7.4"
+    "typescript": "^4.9.5"
   },
   "dependencies": {
-    "ajv": "^8.11.0"
+    "ajv": "^8.11.0",
+    "multiformats": "^9.6.4",
+    "cross-fetch": "^3.1.5",
+    "ipfs-only-hash": "^4.0.0"
```
Comment (anxolin) [inline package.json:56]: Are all of these something we will want to add as dependencies or should some be peer-dependencies?

File: setupTests.js [hunk 3]
```diff
@@ -0,0 +1,10 @@
+import fetchMock from 'jest-fetch-mock'
+import { jest } from '@jest/globals'
+
+global.window = global
+global.jest = jest
```
Comment (anxolin) [inline setupTests.js:5]: why is this needed?

Comment (shoom3301) [reply setupTests.js]: Deleted it, was experimenting with it

File: src/utils.ts (hunk 1 above)
Comment (shoom3301) [reply src/utils.ts]: Deleted

File: package.json (hunk 2 above)
Comment (shoom3301) [reply package.json:56]: Since we lazy load them it's a good idea!

**Comment by shoom3301** [comment]
> Have you tested this anywhere yet, like the Explorer?

https://github.com/cowprotocol/cowswap/pull/2062 - CowSwap
https://github.com/cowprotocol/explorer/pull/352 - Explorer

@alfetopito it works fine with CowSwap, and doesn't work with Explorer:
image

Have you ever faced the problem?

**Comment by shoom3301** [comment]
I found that webpack fails because it "doesn't understand" Optional chaining:
https://github.com/primefaces/primevue/issues/874

**Comment by shoom3301** [comment]
Added `@babel/plugin-proposal-optional-chaining` to Explorer, because it has pretty old version of webpack
